rules_python_generator/wheel_generator.py

```diff
@@ -0,0 +1,479 @@
+
+    def __hash__(self):
+        return hash(self.name)
+
+    def __repr__(self):
+        return self.name
+
+
+class RequirementsToBazelLibGenerator:
+    def __init__(self, requirements_path: str, wheel_dir: str, output_file: str, bzl_path: str):
+        self.requirements_path = requirements_path
+        self.wheel_dir = wheel_dir
+        self.output_file = output_file
+        self.bzl_path = bzl_path
+
+    def _validate(self):
+        with open(self.requirements_path, 'rb') as f:
+            if '--use-wheel' in f.readlines():
+                logger.error('requirements.txt may not contain --use-wheel')
+                sys.exit(1)
+        whl_path = pathlib.Path(self.wheel_dir)
+
+        if not whl_path.parent.exists():
+            logger.error("Wheel dir parent directory '%s' does not exist", whl_path.parent)
+            sys.exit(1)
+
+        output_path = pathlib.Path(self.output_file)
+        if not output_path.parent.exists():
+            logger.error("Output path parent directory '%s' does not exist", output_path.parent)
+            sys.exit(1)
+
+    def _get_wheelname_from_link(self, wheel_link: str) -> str:
+        """ Give a wheel url, return the filename.
+
+        >>> _get_wheelname_from_link('https://foo/__packages/idna_ssl-1.1.0-py3-none-any.whl')
+        "idna_ssl-1.1.0-py3-none-any.whl"
+        """
+        logger.debug('getting filename for %s', wheel_link)
+        match = WHEEL_FILENAME_RE.search(wheel_link)
+        if match:
+            return match.group('filename')
+        else:
+            return ''
+
+    def _get_wheel_links(self, wheel_dir: str) -> dict:
+        wheel_links = {}
+        logger.info('Calling pip wheel on: %s', self.requirements_path)
+        start = time.time()
+        proc = subprocess.Popen(
+            shlex.split(
+                'python3.7 -m pip wheel --verbose --disable-pip-version-check '
+                '--requirement {} --wheel-dir {}'.format(self.requirements_path, wheel_dir)
+            ),
+            stdout=subprocess.PIPE,
+            stderr=subprocess.PIPE,
+            universal_newlines=True,
+        )
+        out, err = proc.communicate()
+        if proc.returncode != 0:
+            logger.exception(err)
+            sys.exit(1)
+        for line in out.splitlines():
+            match = WHEEL_LINK_RE.search(line)
+            if match:
+                # get filename from link and then add to a lookup dict
+                # to store the location of the http links for wheels
+                link = match.group('link')
+                filename = self._get_wheelname_from_link(link)
+                logger.debug('Found link: %s for: %s', link, filename)
+                wheel_links[filename] = link
+        end = time.time()
+        logger.info('pip executed in %s', (end - start) * 1000.0)
+        logger.debug('found: %r', wheel_links)
+        return wheel_links
+
+    def _parse_wheel_dependencies(self, wheel_dir: str,
```

Comment:
Let's add a docstring to these functions as well. Currently, it's kinda confusing, since this function not only parses dependencies, but also downloads them on occasion.